### PR TITLE
[Misc][LoRA] Increase max vocab size limit to 258048 in logits processor

### PR DESCRIPTION
**Pull request by bhoomit**
Increases the LoRA logits processor vocab size upper bound from 257024 to 258048 to support models with larger vocabularies (e.g. 258048 vocab size). Adds test coverage for the new limit.

### CODE REVIEW

File: vllm/lora/layers/logits_processor.py [hunk 1]
```diff
@@ -88,9 +88,9 @@ def create_lora_weights(
         model_config: PretrainedConfig | None = None,
     ) -> None:
         # TODO: Verify if this condition can be further relaxed
-        if 32000 < self.base_layer.vocab_size > 257024:
+        if 32000 < self.base_layer.vocab_size > 258048:
             raise ValueError(
-                "When using LoRA, vocab size must be 32000 >= vocab_size <= 257024"
+                "When using LoRA, vocab size must be 32000 >= vocab_size <= 258048"
             )
```
[the remark this answers is not in the document]
Comment (dcmaddix) [reply vllm/lora/layers/logits_processor.py]: Yes minor I think left inequality sign is wrong direction.

[the remark this answers is not in the document]
Comment (bhoomit) [reply vllm/lora/layers/logits_processor.py]: Thanks @dcmaddix. I have fixed this, as well as the test case.

[the remark this answers is not in the document]
Comment (dcmaddix) [reply vllm/lora/layers/logits_processor.py]: looks good thanks for fixing!

**Comment by robertgshaw2-redhat** [comment]
@alex-jw-brooks mind reviewing this?

File: vllm/lora/layers/logits_processor.py [hunk 2]
```diff
@@ -88,9 +88,9 @@ def create_lora_weights(
         model_config: PretrainedConfig | None = None,
     ) -> None:
         # TODO: Verify if this condition can be further relaxed
-        if 32000 < self.base_layer.vocab_size > 257024:
+        if self.base_layer.vocab_size <= 32000 or self.base_layer.vocab_size > 258048:
```
Comment (alex-jw-brooks) [inline vllm/lora/layers/logits_processor.py:91]: It may be a nice follow-up to make the bounds of this a bit less random so that we don't have to keep bumping it to other seemingly random numbers when a model is over the size 🙂

**Review by alex-jw-brooks — APPROVED** [review]
LGTM, thanks!

File: vllm/lora/layers/logits_processor.py (hunk 2 above)
Comment (bhoomit) [reply vllm/lora/layers/logits_processor.py:91]: I agree, 

Happy to discuss more on how we can make this less arbitrary.

**Review by varun-sundar-rabindranath — APPROVED** [review]
Thanks @bhoomit 

for my own edification, what model requires this ?

**Comment by bhoomit** [comment]
@varun-sundar-rabindranath Thanks for the approval. 

> for my own edification, what model requires this ?

We encountered this with a custom fine-tuned model that has a larger vocabulary than the base model. Previously the lora_extra_vocab_size config flag allowed setting this dynamically, but since it was recently removed, the current hardcoded value is 1024 small to handle the vocab size difference. Bumping it up seemed like the simplest fix for now.

**Review by jeejeelee — APPROVED** [review]
